### PR TITLE
fix: handle @public JSDoc on controllers and quote SDK property names

### PR DESCRIPTION
**Pull request by aakash1122**
## Summary

- **Fix `@public` JSDoc on controller classes**: TypeScript parses `@public` as a built-in visibility tag (`KindJSDocPublicTag`), not a custom `KindJSDocTag`. `extractJSDocTagInfo` was not handling this kind, so class-level `@public` was silently ignored — routes never inherited the public flag from their controller. Added explicit handling for `KindJSDocPublicTag`.
- **Fix unquoted property names in SDK `types.ts`**: The SDK generator emitted property names verbatim in generated interfaces. Names containing spaces, dashes, dots, or starting with digits produced invalid TypeScript (e.g. `first name?: string` instead of `"first name"?: string`). Added `tsPropertyKey()` helper to quote non-identifier names, applied to both `GenerateInterface` and `objectToInlineTS`.
- **Add class-level JSDoc test coverage**: Previously there were zero tests for class-level JSDoc on controllers. Added tests for `@tag`, `@security`, `@hidden`, and `@public` inheritance from controller class to routes, including method-level override behavior.
- **Add e2e tag verification**: The realworld fixture uses class-level `@tag` on all controllers but no e2e test verified the tags appeared in the OpenAPI output. Added assertion checking all 5 expected tags.

## Test plan

- [x] `TestControllerAnalyzer_ClassLevelTag` — class `@tag` inherited by routes (3 subtests: basic, with security+description, method override)
- [x] `TestControllerAnalyzer_ClassLevelSecurity` — class `@security` inherited by routes
- [x] `TestControllerAnalyzer_ClassLevelHidden` — class `@hidden` sets `IgnoreOpenAPI=true`
- [x] `TestControllerAnalyzer_ClassLevelPublic` — class `@public` inherited by routes (was failing before fix)
- [x] `TestGenerateInterface_PropertyWithSpaces` — interface properties with spaces are quoted
- [x] `TestSchemaToTS_InlineObjectWithSpaces` — inline object properties with spaces are quoted
- [x] `TestTsPropertyKey` — unit tests for the identifier check helper
- [x] e2e: realworld fixture OpenAPI tags match class-level `@tag` values
- [x] Full unit test suite passes (`go test ./internal/...`)

### CODE REVIEW

File: internal/sdkgen/schema2ts.go [hunk 1]
```diff
@@ -161,12 +161,33 @@ func GenerateInterface(name string, node *SchemaNode, visited map[string]bool) s
...
 	}
 	sb.WriteString("}\n")
 	return sb.String()
 }
 
+// tsPropertyKey returns a properly quoted TypeScript property key.
+// Valid identifiers are returned as-is, while names containing spaces
+// or other non-identifier characters are double-quoted.
+func tsPropertyKey(name string) string {
+	if len(name) == 0 {
```
Comment (shahriar-shojib) [inline internal/sdkgen/schema2ts.go]: will we ever have empty string properties? if so should we return this?
